test_runner/regress/test_lsn_mapping.py

```diff
@@ -67,3 +69,100 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
             assert endpoint_here.safe_psql("SELECT max(x) FROM foo")[0][0] == i
 
             endpoint_here.stop_and_destroy()
+
+
+# Test pageserver get_time_range_of_lsn API
+def test_ts_of_lsn_api(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+
+    new_timeline_id = env.neon_cli.create_branch("test_ts_of_lsn_api")
+    endpoint_main = env.endpoints.create_start("test_ts_of_lsn_api")
+    log.info("postgres is running on 'test_ts_of_lsn_api' branch")
+
+    cur = endpoint_main.connect().cursor()
+    # Create table, and insert rows, each in a separate transaction
+    # Disable synchronous_commit to make this initialization go faster.
+    #
+    # Each row contains current insert LSN and the current timestamp, when
+    # the row was inserted.
+    cur.execute("SET synchronous_commit=off")
+    cur.execute("CREATE TABLE foo (x integer)")
+    tbl = []
+    for i in range(1000):
+        cur.execute("INSERT INTO foo VALUES(%s)", (i,))
+        # Get the timestamp at UTC
+        after_timestamp = query_scalar(cur, "SELECT clock_timestamp()").replace(tzinfo=timezone.utc)
+        after_lsn = query_scalar(cur, "SELECT pg_current_wal_lsn()")
+        tbl.append([i, after_timestamp, after_lsn])
+
+    # Execute one more transaction with synchronous_commit enabled, to flush
+    # all the previous transactions
+    cur.execute("INSERT INTO foo VALUES (-1)")
+
+    # Wait until WAL is received by pageserver
+    wait_for_last_flush_lsn(env, endpoint_main, env.initial_tenant, new_timeline_id)
+
+    last_flush_lsn = Lsn(query_scalar(cur, "SELECT pg_current_wal_flush_lsn()"))
+
+    with env.pageserver.http_client() as client:
+        # Check edge cases: lsn larger than the last flush lsn
+        probe_lsn = Lsn(int(last_flush_lsn) * 20 + 80_000)
+        result = client.timeline_get_time_range_of_lsn(
+            env.initial_tenant,
+            new_timeline_id,
+            probe_lsn,
+        )
+        # assert result == "future"
```

Comment:
```suggestion
        assert result == "future"
```

Or is this intentional?

Reply:
also copy paste error